File: Core/src/main/java/com/github/ezauton/core/actuators/VelocityProcessor.java
package com.github.ezauton.core.actuators;

/**
 * A class which takes in a velocity, and automatically applies the velocity to another motor
 * This is nice for grouping multiple velocity processors on top of each other. Primarily used for simulated motors.
 */
public abstract class VelocityProcessor implements VelocityMotor {
    private final VelocityMotor toApply;

    public VelocityProcessor(VelocityMotor toApply) {
        this.toApply = toApply;
    }

    public VelocityMotor getToApply() {
        return toApply;
    }

}

File: Core/src/main/kotlin/com/github/ezauton/core/actuators/VelocityProcessor.kt
package com.github.ezauton.core.actuators

/**
 * A class which takes in a velocity, and automatically applies the velocity to another motor
 * This is nice for grouping multiple velocity processors on top of each other. Primarily used for simulated motors.
 */
abstract class VelocityProcessor(val toApply: RotVelMotor) : RotVelMotor
